rpc/core/tx.go

```diff
@@ -113,15 +125,69 @@ func TxSearch(
 	for i := skipCount; i < skipCount+pageSize; i++ {
 		r := results[i]
 
+		var txProof types.TxProof
+		if prove {
+			txProof, err = proveTx(r.Height, r.Index)
+			if err != nil {
+				return nil, err
+			}
+		}
+
 		apiResults = append(apiResults, &ctypes.ResultTx{
 			Hash:     types.Tx(r.Tx).Hash(),
 			Height:   r.Height,
 			Index:    r.Index,
 			TxResult: r.Result,
 			Tx:       r.Tx,
-			Proof:    types.TxProof{}, // transaction inclusion proofs are not currently supported by this endpoint
+			Proof:    txProof,
 		})
 	}
 
 	return &ctypes.ResultTxSearch{Txs: apiResults, TotalCount: totalCount}, nil
 }
+
+func proveTx(height int64, index uint32) (types.TxProof, error) {
+	var (
+		pTxProof tmproto.TxProof
+		txProof  types.TxProof
+	)
+	rawBlock, err := loadRawBlock(env.BlockStore, height)
+	if err != nil {
+		return txProof, err
+	}
+	res, err := env.ProxyAppQuery.QuerySync(abcitypes.RequestQuery{
```

Comment:
[Non Blocking][Question]
Can you please link me where the proof is created?

Reply:
yeah sure, the finishing touches are being done on https://github.com/celestiaorg/celestia-app/tree/evan/query-tx-inclusion-proofs

unfortunately, since we've delete the share encoding code here, there was still some legacy code (the code for PrepareProposal that doesn't use non-interacitve defaults) that has broke after switching to the new code (to be expected I think). So still fixing that before we can actually use this code and test it.

Reply:
Awesome, thanks a lot